### ISSUE TITLE
Create host and controller submodules

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#290]
## Summary

It can be difficult to understand which handlers belong to host and which to controller. The keeper package should have two files, host.go and controller.go. For example, `InitInterchainAccount` should go in `controller.go`. I'm fine with moving or leaving the handshakes as well

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by damiannolan** [comment]
[Rename app root pkg definition also](https://github.com/cosmos/ibc-go/pull/417#discussion_r743541950)

**Comment by AdityaSripal** [comment]
Agreed with this approach. I do the same thing for CCV. The parent and child modules are separated rather than keeping it in the same ibc module.

https://github.com/cosmos/interchain-security

**Comment by damiannolan** [comment]
closed by #541

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
